UPGRADING.md

```diff
@@ -0,0 +1,516 @@
+import { ensureReady, After } from '@jaredpalmer/after';
+import './client.css';
+import routes from './routes';
+
+ensureReady(routes).then(data =>
+  hydrate(
+    <BrowserRouter>
+      <After data={data} routes={routes} />
+    </BrowserRouter>,
+    document.getElementById('root')
+  )
+);
+
+if (module.hot) {
+  module.hot.accept();
+}
+```
+
+To:
+
+```jsx
+// client.js
+
+import React from 'react';
+import { hydrate } from 'react-dom';
+import { BrowserRouter } from 'react-router-dom';
+import { After } from '@jaredpalmer/after';
+import './client.css';
+import routes from './routes';
+
+hydrate(
+  <BrowserRouter>
+    <After routes={routes} />
+  </BrowserRouter>,
+  document.getElementById('root')
+);
+
+if (module.hot) {
+  module.hot.accept();
+}
+```
+
+You can access server app state from the window (`window.__SERVER_APP_STATE__`), you can modify `SERVER_APP_STATE` and pass it to `<After />` component.
+
+```jsx
+<After data={modifyData(window.__SERVER_APP_STATE__)} routes={routes} />
+```
+
+`After` component will pick `window.__SERVER_APP_STATE__` if data prop not passed.
+
+```jsx
+function After({ routes, data = window.__SERVER_APP_STATE__ }) {
+  // ...
+}
+```
+
+### `routes.js` config changed
+
+To send assets (CSS and JS files) from the initial server response, we need to change how we define our routes.
+
+From:
+
+```jsx
+// routes.js
+
+import Home from './Home';
+import { asyncComponent } from '@jaredpalmer/after';
+
+export default [
+  {
+    path: '/',
+    exact: true,
+    component: Home,
+  },
+  {
+    path: '/about',
+    exact: true,
+    component: asyncComponent({
+      loader: () => import('./About'),
+    }),
+  },
+  {
+    path: '/contact-us',
+    exact: true,
+    component: asyncComponent({
+      loader: () => import('./Contact'),
+    }),
+  },
+];
+```
+
+To:
+
+```jsx
+// routes.js
+
+import Home from './Home';
+import { asyncComponent } from '@jaredpalmer/after';
+
+export default [
+  {
+    path: '/',
+    exact: true,
+    component: Home,
+  },
+  {
+    path: '/about',
+    exact: true,
+    component: asyncComponent({
+      loader: () => import(/* webpackChunkName: "whatever" */ './About'),
+      chunkName: 'whatever',
+    }),
+  },
+  {
+    path: '/contact-us',
+    exact: true,
+    component: asyncComponent({
+      loader: () => import(/* webpackChunkName: "ContactUs" */ './Contact'),
+      chunkName: 'ContactUs',
+    }),
+  },
+];
+```
+
+> This is too hard and complicated (and there are more things that we have to take care about) so we made a babel plugin to do this automatically because we care about you (DX).
+
+#### Use babel plugin
+
+Create a `.babelrc` file in the root of the project (next to the package.json)
+
+```js
+// .babelrc
+
+{
+  "presets": [ "razzle/babel" ],
+  "plugins": [ "after-async-component" ]
+}
+```
+
+> Check [after-async-component](https://github.com/nimacsoft/babel-plugin-after-async-component) repo for more options.
```

Comment:
Move the babel plugin to this repo.

Reply:
should I create a folder called `packages` in the root of this repo? 
```
-
 src/
 packages/
   babel-plugin/ <- should i call this folder `babel-plugin` or something else ?
```

Reply:
call the folder `babel-plugin-after`

Reply:
done